WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs

```diff
@@ -20,84 +21,105 @@ public class LoginViewModel : INotifyPropertyChanged
-            SavePersonCommand = new Command(SavePerson);
-            TakePhotoCommand = new Command(TakePhoto);
-            PersonObject = new ImageObject();
-            _restService = new RestService();
-        }
-        public bool ErrorLabel { get; set; }
+        
         public INavigation Navigation { get; set; }
-
+        
         public event PropertyChangedEventHandler PropertyChanged;
+        
+        private RestService _restService;
+        
+        public bool ErrorLabel { get; set; }
 
         private MediaFile takenPhoto { get; set; }
-        private ImageObject personObject;
+        private ImageObject _personObject;
+        
+        //Later on will map to display on frontend
+        public string ErrorMessage { get; set; } = "Please fill in all the fields!";
 
         public ImageObject PersonObject
         {
             get
             {
-                return personObject;
+                return _personObject;
             }
             set
             {
-                personObject = value;
+                _personObject = value;
                 //OnPropertyChanged();
             }
         }
+        
+        public LoginViewModel()
+        {
+            SavePersonCommand = new Command(SavePerson);
+            TakePhotoCommand = new Command(TakePhoto);
+            PersonObject = new ImageObject();
+            _restService = new RestService();
+        }
 
         public async void TakePhoto()
         {
             //Checking for camera permissions
-            bool cameraPermission = await PermissionHandler.CheckForCameraPermission();
+            var cameraPermission = await PermissionHandler.CheckForCameraPermission();
             if (!cameraPermission)
                 await CrossPermissions.Current.RequestPermissionsAsync(Permission.Camera);
 
             //Checking for storage permissions
-            bool storagePermission = await PermissionHandler.CheckForCameraPermission();
+            var storagePermission = await PermissionHandler.CheckForCameraPermission();
             if (!storagePermission)
                 await CrossPermissions.Current.RequestPermissionsAsync(Permission.Storage);
 
             //Taking photo and storing it in MediaFile variable 'takenPhoto'
-
             takenPhoto = await TakingPhotoHandler.TakePhoto();
-
         }
 
         public async void SavePerson()
         {
-            //Checks if all required fields are filled
             if (!FieldsAreFilled())
             {
                 return;
             }
 
-            //If so, stores his information in the db
-            /*PersonObject.ImageName = PersonObject.PersonFirstName + PersonObject.PersonLastName + ".jpg";
-            await CloudStorageService.SaveBlockBlob(takenPhoto,PersonObject.ImageName);
-            PersonObject = await _restService.CreateImageObject(PersonObject);*/
+            PersonObject.ImageName = PersonObject.PersonFirstName + PersonObject.PersonLastName + ".jpg";
```

Comment:
Hmm, that is not a particularly safe naming strategy

Reply:
I know, a slight problem here. I wanted to name it with unique ID, but small problem: ID is only generated after inserting element into DB, so if I wanted to use completely safe and unique naming I would need to write sort of update method for image name and was too lazy, again :D, for that. 

After all, these changes were just quick fix for previous pull request, will work on everything a lot more, I want it to be beautiful :D